File: FRONTEND/facturador-frontend/src/services/ApiSucursalService.js
import Api from '@/services/Api'

export default {
    ///ApiSucursal
    getApiSucursal(){
        return Api().get("/api/apiSucursal");
    },
    getApiSucursalPorIdEmpresa(idEmpresa){
        return Api().get(`/api/apiSucursal/listarPorIdEmpresa/${idEmpresa}`)
    },
    listaBySucursalNoTieneCuis(idEmpresa){
        return Api().get(`/api/apiSucursal/listaBySucursalNoTieneCuis/${idEmpresa}`)
    },
    listaByPuntoVentaNoTieneCuis(idEmpresa){
        return Api().get(`/api/apiSucursal/listaByPuntoVentaNoTieneCuis/${idEmpresa}`)
    },
    listaByEmpresaTieneCuisVigente(idEmpresa){
        return Api().get(`/api/apiSucursal/listaByEmpresaTieneCuisVigente/${idEmpresa}`)
    },
    postApiSucursal(dato){
        return Api().post("/api/apiSucursal", dato);
    },
    putApiSucursal(dato){
        return Api().put("/api/apiSucursal", dato);
    },

}

File: BACKEND/facturador_backend/src/main/java/com/bap/api/services/api/ApiSucursalService.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package com.bap.api.services.api;

import com.bap.api.configurate.AbstractEntity;
import com.bap.api.model.api.ApiSucursal;
import java.util.List;

/**
 *
 * @author ruth
 */
public interface ApiSucursalService extends AbstractEntity<ApiSucursal> {

    List<ApiSucursal> listarPorIdEmpresa(Long idEmpresa);    
    
    List<ApiSucursal> listarByNoTieneCuis(Long idEmpresa);
    
    List<ApiSucursal> listaByPuntoVentaNoTieneCuis(Long idEmpresa);
    
    List<ApiSucursal> listaByEmpresaTieneCuisVigente(Long idEmpresa);
    
    ApiSucursal leerPorIdEmpresaAndCodigoSucursal(Long idEmpresa, Long codigoSucursal);
    
}
